truss/util/config_checks.py

```diff
@@ -0,0 +1,41 @@
+import requests
+from truss.config.trt_llm import CheckpointSource
+from truss.constants import (
+    HF_MODELS_API_URL,
+    TRTLLM_MIN_MEMORY_REQUEST,
+    TRTLLM_MIN_MEMORY_REQUEST_BYTES,
+)
+from truss.truss_handle import TrussHandle
+
+
+def check_secrets_for_trt_llm_builder(tr: TrussHandle) -> bool:
+    if tr.spec.config.trt_llm and tr.spec.config.trt_llm.build:
+        source = tr.spec.config.trt_llm.build.checkpoint_repository.source
+        hf_model_id = tr.spec.config.trt_llm.build.checkpoint_repository.repo
+        if (
+            source == CheckpointSource.HF
+            and "hf_access_token" not in tr.spec.secrets
+            and not _is_model_public(hf_model_id)
+        ):
+            return False
+    return True
+
+
+def check_and_update_memory_for_trt_llm_builder(tr: TrussHandle) -> None:
+    if tr.spec.config.trt_llm and tr.spec.config.trt_llm.build:
+        if (
+            not tr.spec.memory_in_bytes
+            or tr.spec.memory_in_bytes < TRTLLM_MIN_MEMORY_REQUEST_BYTES
+        ):
```

Comment:
is there ever a valid case where the `tr.spec.memory` is set but `tr.spec.memory_in_bytes` is not set? i.e. could we potentially overwrite user memory requests with this conditional?

Reply:
Since we go through the `from_yaml` method to create the config, there shouldn't be - the memory spec will always be parsed, and the spec was invalid in the config then loading it would've failed before
but, I don't love that memory_in_bytes being an (optional) member resources, when it's just a derived property. i think i'm going move the memory spec parser to a helper and then call that from the `memory_in_bytes` property

Reply:
yeah that sounds good to me too – thanks for explaining